gateway/gateway_test.go

```diff
@@ -60,44 +86,681 @@ func (m mockNamesys) GetResolver(subs string) (namesys.Resolver, bool) {
 	return nil, false
 }
 
-type mockApi struct {
-	ns mockNamesys
+type mockAPI struct {
+	blockStore   blockstore.Blockstore
+	blockService blockservice.BlockService
+	dagService   format.DAGService
+	resolver     resolver.Resolver
+	namesys      mockNamesys
 }
 
-func newMockApi() *mockApi {
-	return &mockApi{
-		ns: mockNamesys{},
+func newMockAPI(t *testing.T) (*mockAPI, cid.Cid) {
+	r, err := os.Open("./testdata/fixtures.car")
```

Comment:
@hacdias fysa I've moved fixtures to `testdata/` subdir, it seems to be a good convention to follow in future: 

> The go tool will ignore a directory named "testdata", making it available to hold ancillary data needed by the tests.
> – `go help test`